### PR TITLE
feat: remove legacy style components

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Remove all the code related to deprecated stylesheet in @stream-io/stream-chat-css.

Depends on https://github.com/GetStream/stream-chat-css/pull/296

### 🛠 Implementation details

1. Drop `themeVersion` property from ChatContext/ChatProvider ✅

2. Drop all of the markup which relies on `themeVersion` 
- Audio.tsx - drop or rename V1 component to AudioLegacy ✅
- Card.tsx - drop or rename V1 component to CardLegacy ✅
- FileAttachment.tsx - drop or rename V1 component to FileAttachmentLegacy ✅
- ChannelSearch.tsx - remove showSearchBarV2 condition ✅
- SearchResults.tsx - remove conditions, adjust markup ✅
- EmojiPicker.tsx - remove conditions and associated classNames, drop related icons ✅
- MessageRepliesCountButton.tsx - remove condition and related icon ✅
- MessageSimple.tsx - drop conditions and V1 markup ✅
- MessageStatus.tsx - drop conditions and V1 markup ✅
- EditMessageForm.tsx - drop conditions and V1 markup ✅
- MessageInputFlat.tsx - drop MessageInputV1 ✅
- QuotedMessagePreview.tsx - drop conditions and V1 markup ✅
- SendButton.tsx - drop conditions and V1 markup ✅
- MessageListMainPanel.tsx - drop conditions and V1 markup ✅
- Modal.tsx - drop conditions and V1 markup ✅
- SimpleReactionsList.tsx - drop conditions and V1 markup ✅
- SimpleReactionsList.tsx - (possibly remove the whole component?)
- Thread.tsx - drop conditions and V1 markup ✅
- TypingIndicator.tsx - drop conditions and V1 markup ✅
- UserItem.tsx - drop conditions and V1 markup ✅
- Avatar.tsx - adjust to CSS vars (like in this [[component](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Reactions/StreamEmoji.tsx)](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Reactions/StreamEmoji.tsx)) ✅
- MessageInputSmall.tsx - remove file ✅
- ReactionSelector.tsx - should we remove `detailedView` prop if markup displayed with it is not styled in v2? ⁉️

3. remove V1-related `classNames` from components which work with both versions ✅

4. drop theme-V1-related deprecated components 
- Tooltip.tsx -> Tooltip (?) - FileUploadButton.tsx + AttachmentIcon.tsx ✅
- FileIcon.tsx -> drop version property and V1 icons ✅
- UploadsPreview.tsx - MessageInputSmall.tsx ✅
- MessageInput/icons.tsx -> EmojiIconLarge ✅
- MessageInput/icons.tsx -> EmojiIconSmall ✅
- MessageInput/icons.tsx -> FileUploadIcon ✅
- MessageInput/icons.tsx -> FileUploadIconFlat ✅
- MessageInput/icons.tsx -> SendIconV1 ✅
- MessageList/icons.tsx -> ArrowUp - will keep as complementary to ArrowDown
- Message/icons.tsx -> ReplyIcon ✅
- Message/icons.tsx -> DeliveredCheckIcon ✅
- Message/icons.tsx -> ErrorIcon ✅
- Message/icons.tsx -> PinIcon - will keep for those who want to implement pinned messages list 🚫
- Message/icons.tsx -> PinIndicator  - will keep for those who want to implement pinned messages list 🚫

5. Fix bug in MessageInputFlat - not allowing to retry uploads as numberOfUploads does not include failed uploads ✅
6. remove theme V1 CSS from final bundle and exports, move everything related to v2 styling from `dist/css/v2` folder to `dist/css`, drop v2 folder ✅
7. v12 docs & drop [[Legacy](https://getstream.io/chat/docs/sdk/react/guides/theming/css_and_theming/)](https://getstream.io/chat/docs/sdk/react/guides/theming/css_and_theming/) section ✅
8. v12 migration guide ✅
9. Review docs for references to removed components ✅
10. Remove `Chat` component prop `customStyles` and associated code ✅
11. Change docs theming/introduction.mdx id from themingv2 to something else - will not be done as we do not want to change URLs and do not have means to redirect 🚫
12. drop tests related to v1 components & markup ✅
13. rename V2 components (CardV2 → Card) ✅
14. Remove ChatDown component ✅
15. Remove `$stream-chat-theme-version: '2';` from example apps’ CSS ✅

### Out of scope
1. Remove `--str-chat__theme-version` from CSS docs - global-variables.mdx - needs change in generation scripts
2. Remove FixedHeightMessage?
3. Remove theme prop from MessageText, MessageOptions?

BREAKING CHANGE: stylesheet import path changed & v1 stylesheet has been dropped, see [release guide](...) for more information
BREAKING CHANGE: theme v1 related markup and classNames have been removed
BREAKING CHANGE: `themeVersion` property has been removed from `ChatContext`

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -83,10 +83,8 @@ const basePlugins = ({ useBrowserResolve = false }) => [
   copy({
     targets: [
       { dest: 'dist/assets', src: './node_modules/@stream-io/stream-chat-css/dist/assets/*' },
-      { dest: 'dist/css', src: './node_modules/@stream-io/stream-chat-css/dist/css/*.css' },
-      { dest: 'dist/scss', src: './node_modules/@stream-io/stream-chat-css/dist/scss/*' },
-      { dest: 'dist/css/v2', src: './node_modules/@stream-io/stream-chat-css/dist/v2/css/*.css' },
-      { dest: 'dist/scss/v2', src: './node_modules/@stream-io/stream-chat-css/dist/v2/scss/*' },
+      { dest: 'dist/css', src: './node_modules/@stream-io/stream-chat-css/dist/v2/css/*.css' },
+      { dest: 'dist/scss', src: './node_modules/@stream-io/stream-chat-css/dist/v2/scss/*' },
```
Comment (oliverlaz) [inline rollup.config.js:87]: I think we should just remove the legacy theming from the bundle and keep the modern theming under `v2/css`. That way, this won't be a breaking change for customers who are already on v2 theming.

Comment (arnautov-anton) [reply rollup.config.js:87]: I know about reducing the amount of breaking changes per such release but in this case I think leaving behind `v2` traces seems sub-optimal. Let's clean it all up while we're at it and as this is pretty noticeable breaking change, we can lead with it in the release guide. :)

Comment (oliverlaz) [reply rollup.config.js:87]: Update: after discussing this internally, we agreed to remove the `/v2` from the path.

Comment (MartinCupela) [reply rollup.config.js:87]: Update: after another discussion, we decided to keep `v2` in the path as there are even video tutorials, but also sandboxes and written tutorials using this path.

Comment (MartinCupela) [reply rollup.config.js]: Ref to commit: https://github.com/GetStream/stream-chat-react/pull/2394/commits/c7caaa2b88c8ac9cfd21720fc20261e21d703c27

**Review by oliverlaz — APPROVED** [review]

**Pull request merged**